apps/visualizer/src/pages/home.tsx

```diff
@@ -10,31 +10,51 @@ export default function Home() {
 
   const [errorMessage, setError] = createSignal<string | null>(null)
 
-  async function handleDrop(e: DragEvent) {
+  function handleDrop(e: DragEvent) {
     e.preventDefault()
+    processFile(e.dataTransfer?.files || new FileList())
+  }
 
-    const files = Array.from(e.dataTransfer.files)
+  function handleDragOver(e: DragEvent) {
+    e.preventDefault()
+  }
 
-    const transactions = (
-      await Promise.all(files.map(processStatementFile))
-    ).flat()
+  const headers = ["Amount", "Payment Date"]
 
-    setTransactions(transactions)
+  function handleClickUploadFile(e: MouseEvent) {
+    e.preventDefault()
+    document.getElementById("statement-input")?.click()
```

Comment:
Recommendation is to refactor this to attach refs with `ref={input}`, so it does not rely on getElementById krub

https://www.solidjs.com/tutorial/bindings_refs

Reply:
Thanks krub! Almost forgot about `ref `